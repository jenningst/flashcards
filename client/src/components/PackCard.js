import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { withRouter, Link } from 'react-router-dom';

import { Title4 } from './elements/Text';

function PackCard({ _id, name }) {
  return (
    <LinkWrapper to={`/pack/${_id}`}>
      <OuterWrapper>
        <PackCardContent
          className='PackCard'
          data-testid='card'
          type='button'
        >
          <Title4
            className='PackCard__title'
          >
            {name}
          </Title4>
        </PackCardContent>
      </OuterWrapper>
    </LinkWrapper>
  );
};

const LinkWrapper = styled(Link)`
  text-decoration: none;
`;

const BaseWrapper = styled.div`
  box-sizing: border-box;
  border-radius: 1rem;
`;

const OuterWrapper = styled(BaseWrapper)`
  background: #6E78CE;
`;

const PackCardContent = styled(BaseWrapper)`
  display: flex;
  justify-content: center;
  align-items: center;
  background: ${props => props.theme.color.main.primary};
  color: ${props => props.theme.color.fonts.offWhite};
  
  margin-right: .50rem;
  padding: .70rem;
  min-height: 100px;
  border: none;

  &:hover {
    background-color: ${props => props.theme.color.main.primaryHover};
  }

  h4 {
    text-align: center;
    font-weight: 400;
  }
`;

PackCard.propTypes = {
  _id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default withRouter(PackCard);