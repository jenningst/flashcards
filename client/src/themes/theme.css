/* 

button: {
      primary: '#7883E2',
      primaryHover: '#939CE8',
      active: '#5860A5',
      disabled: '#e0e0eb',
    },
    font: {
      pureWhite: '#ffffff',
      offWhite: '#fcfbfc',
      eerieBlack: '#16171D',
      charleston: '#292b3d',
      grey: '#b1aeb7',
      lightGrey: '#eff0f1'
    },
    border: {
      primary: '#42487C',
      active: '#939CE8'
      disabled: '#b1aeb7',
      input: '#e9eaf0',
    }, // TODO: consolidate everything below; using the above
    main: {
      primary: '#7883E2',
      primaryHover: '#939CE8',
      secondary: '#e3e5ff',
      secondaryHover: '#c3d2ff',
      pureWhite: '#ffffff',
      offWhite: '#f5f7fb',
    },
    alerts: {
      notice: '#c0a5ff',
      success: '#50cdff',
      successHover: '#6FD6FF',
      warning: '#FFCB17',
      warningBg: '#FFDA5C',
      danger: '#DE6A60',
    },
    fonts: {
      eerieBlack: '#16171D',
      charleston: '#292b3d',
      pureWhite: '#ffffff',
      offWhite: '#fcfbfc',
      grey: '#b1aeb7',
      lightGrey: '#eff0f1'
    },
    borders: {
      lightGrey: '#d8d6db',
    }

}*/