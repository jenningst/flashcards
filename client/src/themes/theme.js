export const darkTheme = {
  font: {
    primary: '#FFFFFF',
    secondary: '#E0E0E0',
    tertiary: '#666666',
  },
  background: {
    primary: '#282626',
    secondary: '#313131',
    highlight: '#6121BF',
    special: '#7358F7',
    inactive: '#b1aeb7',
  },
  button: {
    default: {
      primary: '#7358F7',
      border: '#4D18AE',
      alert: '#D55949',
      greyed: '#666666',
    },
    disabled: {
      primary: '',
      secondary: '',
      border: '',
    },
    hover: {
      primary: '#6121BF',
      border: '#6121BF',
    },
    active: {
      primary: '#8b90ff',
      border: '',
    },
  },
};

export const lightTheme = {
  color: {
    button: {
      primary: '#939CE8',
      primaryHover: '#7883E2',
      active: '#5860A5',
      secondary: '#50cdff',
      secondaryHover: '#6FD6FF',
      disabled: '#e0e0eb',
    },
    font: {
      pureWhite: '#ffffff',
      offWhite: '#fcfbfc',
      eerieBlack: '#16171D',
      charleston: '#33364d',
      blackCoral: '#585A6D',
      grey: '#b1aeb7',
      lightGrey: '#eff0f1',
      placeholder: '#e9eaf0',
      danger: '#DE6A60',
    },
    border: {
      primary: '#5860A5',
      secondary: '#7883E2',
      disabled: '#b1aeb7',
      inputDefault: '#e9eaf0',
      transparent: '#0000ffff',
      eerieBlack: '#16171D',
      charleston: '#33364d',
    }, // TODO: consolidate everything below; using the above
    background: {
      pureWhite: '#ffffff',
      offWhite: '#fcfbfc',
    },
    main: {
      primary: '#7883E2',
      primaryHover: '#939CE8',
      secondary: '#e3e5ff',
      secondaryHover: '#c3d2ff',
      pureWhite: '#ffffff',
      offWhite: '#f5f7fb',
    },
    alerts: {
      notice: '#c0a5ff',
      success: '#50cdff',
      successHover: '#6FD6FF',
      warning: '#FFCB17',
      warningBg: '#FFDA5C',
      danger: '#DE6A60',
    },
    fonts: {
      eerieBlack: '#16171D',
      charleston: '#292b3d',
      pureWhite: '#ffffff',
      offWhite: '#fcfbfc',
      grey: '#b1aeb7',
      lightGrey: '#eff0f1'
    },
    borders: {
      lightGrey: '#d8d6db',
    }
  },
  spacing: {
    compact: '1rem',
    comfortable: '1.5rem',
    generous: '2rem',
  },
  font: {
    primary: '#4e5167',
    secondary: '#262626',
    tertiary: '#666666',
  },
  background: {
    primary: '#ffffff',
    secondary: '#fcfbfc',
    highlight: '#9388ff',

    special: '#50cdff',
    inactive: '#b1aeb7',
  },
  button: {
    default: {
      primary: '#7358F7',
      border: '#4D18AE',
      alert: '#D55949',
      greyed: '#666666',
    },
    disabled: {
      primary: '',
      secondary: '',
      border: '',
    },
    hover: {
      primary: '#6121BF',
      border: '#6121BF',
    },
    active: {
      primary: '#8b90ff',
      border: '',
    },
  },
};