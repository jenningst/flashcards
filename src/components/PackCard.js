import React, { useContext } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import CollectionContext from '../contexts/collectionContext';
import ThemeContext from '../contexts/themeContext';

import { Title4, Caption3 } from '../components/Elements/Text';

PackCard.propTypes = {
  props: PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string,
    count: PropTypes.number.isRequired,
    startCollection: PropTypes.func.isRequired,
  }),
};

function PackCard(props){
  const { name, image, count, startCollection } = props;
  // Context
  const { dispatch } = useContext(CollectionContext);
  const theme = useContext(ThemeContext);
  // Actions
  const setCollection = name => dispatch({ type: 'SET_COLLECTION', name });

  const PackCardWrapper = styled.button`
    padding: .70rem;
    background: ${theme.background.secondary};
    border-radius: 1rem;
    min-height: 130px;
    border: none;
  
    .PackCard__header {
      color: ${theme.font.primary};
      margin-bottom: .50em;
    }
  
    .PackCard__count-pill {
      background: ${theme.background.highlight};
      color: ${theme.font.primary};
      padding: .25rem;
      border-radius: 30rem;
    }
  `;

  return (
    <PackCardWrapper
        className="PackCard"
        type="button"
        onClick={e => setCollection(name)}
      >
        <Title4 className="PackCard__header">{name}</Title4>
        <div className="PackCard__count-pill">
          <Caption3 className="PackCard__count-text">{count} CARDS</Caption3>  
        </div>
      </PackCardWrapper>
  );
};


export default PackCard;