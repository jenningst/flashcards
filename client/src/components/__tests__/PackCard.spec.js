import React from 'react';
import { MemoryRouter, BrowserRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { darkTheme, lightTheme } from '../../themes/theme';

import {
  cleanup,
  render,
} from '@testing-library/react';

import PackCard from '../PackCard';

afterEach(cleanup);

const renderComponent = ({ name, _id, theme }) => 
  render(
    <ThemeProvider theme={theme}>
      <MemoryRouter initialEntries={['/random-page']}>
        <BrowserRouter>
          <PackCard _id={_id} name={name} theme={theme}/>
        </BrowserRouter>
      </MemoryRouter>
    </ThemeProvider>
  );

describe('<PackCard /> spec', () => {
  it('assert component matches snapshot', () => {
    const { asFragment } = renderComponent({ 
      name: 'JavaScript', 
      _id: '1',
      theme: lightTheme });

    expect(asFragment()).toMatchSnapshot();
  });

  it('assert the primary button is rendered', async () => {
    const initialName = 'Vue';
    const _id = '1';
    const { getByTestId } = renderComponent({ 
      name: initialName, 
      _id, 
      theme: lightTheme });

    expect(await getByTestId('card')).toBeInTheDocument();
  });

  it('assert card name is displayed', async () => {
    const initialName = 'JavaScript';
    const _id = '1';
    const theme = lightTheme;
    const { getByText } = renderComponent({ 
      name: initialName, 
      _id, 
      theme: lightTheme });

    expect(await getByText(initialName)).toBeInTheDocument();
  });

  it('assert re-renders with new props', async () => {
    const initialName = 'JavaScript';
    const newName = 'Angular';
    const theme = lightTheme;
    const _id = '1';
    const { getByText, rerender } = renderComponent({ 
      name: initialName, 
      _id, 
      theme: lightTheme });

    expect(await getByText(initialName)).toBeInTheDocument();
    rerender(
      <ThemeProvider theme={lightTheme}>
        <MemoryRouter>
          <PackCard _id={_id} name={newName} theme={theme}/>
        </MemoryRouter>
      </ThemeProvider>
    );
    const newTitle = getByText(newName);

    expect(await newTitle).toBeInTheDocument();
  });
});