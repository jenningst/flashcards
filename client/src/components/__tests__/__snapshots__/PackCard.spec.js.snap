// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackCard /> spec assert component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="PackCard sc-iwsKbI cPNTob"
    data-testid="card"
    type="button"
  >
    <h4
      class="PackCard__title sc-bxivhb dSXQHY"
    >
      JavaScript
    </h4>
  </button>
</DocumentFragment>
`;
